osquery/tables/system/posix/apt_sources.cpp

```diff
@@ -19,51 +19,95 @@
+  auto comment_pos = input_line.find("#");
+  std::string line = input_line;
+  if (comment_pos != std::string::npos) {
+    line = input_line.substr(0, comment_pos);
   }
 
-  // The source name could set [arch=ARCH].
-  size_t offset = (comp.size() > 1 && comp[0][0] == '[') ? 1 : 0;
-  // Seek to the end of the schema.
-  auto host = comp[offset].find("://");
+  // Only look for deb names (not deb-src).
+  auto deb_pos = line.find("deb ");
+  if (deb_pos == std::string::npos) {
+    return Status::failure("No deb prefix");
+  }
+  line = line.substr(deb_pos + 4); // for "deb " length
+
+  // Split on whitespace
+  // additional leading whitespace will get clobbered by split
+  auto tokens = osquery::split(line, " ");
+  if (tokens.empty()) {
+    return Status::failure("Empty line");
+  }
+  if (tokens.size() < 2) {
+    return Status::failure("not enough tokens specified");
+  }
+
+  size_t offset = 0;
+
+  // The source name could set [arch=ARCH option2=something ].
+  if (tokens[offset][0] == '[') {
+    // Seek to the end of the options
+    for (size_t i = offset; i < tokens.size(); i++) {
+      if (tokens[i].find("]") != std::string::npos) {
+        offset = i;
+        break;
+      }
+    }
+    // go to the next token after the close of options ']'
+    offset++;
+  }
+  if (offset >= tokens.size()) {
+    return Status::failure("incomplete line no suite");
+  }
+
+  auto host = tokens[offset].find("://");
   if (host == std::string::npos) {
     return Status::failure("Cannot find protocol");
   }
 
-  apt_source.base_uri = comp[offset];
-  apt_source.cache_file.push_back(comp[offset].substr(host + 3));
-  if (apt_source.cache_file[0].empty()) {
-    return Status::failure("Cache file is empty");
+  apt_source.base_uri = tokens[offset];
+  // Cannot have trailing slashes
+  while (apt_source.base_uri.back() == '/') {
+    apt_source.base_uri.pop_back();
+  }
+
+  // go onto parse the suite
+  offset++;
+  if (offset >= tokens.size()) {
+    return Status::failure("incomplete line no suite");
   }
+  std::string suite = tokens[offset];
 
-  // Cannot have trailing slashes
-  while (apt_source.cache_file.back().back() == '/') {
-    apt_source.cache_file.back().pop_back();
+  // get the target of the uri for the name
+  if (apt_source.base_uri.size() < host + 3) {
+    return Status::failure("empty uri");
   }
+  apt_source.name = apt_source.base_uri.substr(host + 3); // remove "://"
+  // include target uri in cache name
+  apt_source.cache_file.push_back(apt_source.name);
+
+  // Construct the source 'name' from uri and suite.
+  apt_source.name += " " + suite;
 
   bool use_dists = true;
-  for (size_t i = offset + 1; i < comp.size(); i++) {
-    if (comp[i][0] == '#') {
-      // Stop parsing if there is a comment.
-      break;
-    }
-    auto parts = osquery::split(comp[i], "/");
-    use_dists = parts.size() == 1;
+  auto suite_parts = osquery::split(suite, "/");
+  use_dists = suite_parts.size() == 1;
```

Comment:
Can you define `use_dists` here, instead of line 98? 
```cpp
bool use_dists = suite_parts.size() == 1;
```

Reply:
i refactored it to remove use_dists